net_connection/src/json_protocol.rs

```diff
@@ -102,75 +100,189 @@ pub struct FailureResultData {
     pub error_info: serde_json::Value,
 }
 
+//--------------------------------------------------------------------------------------------------
+// DHT Data
+//--------------------------------------------------------------------------------------------------
+
+/// Data Request from own p2p-module
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson)]
-pub struct GetDhtData {
+pub struct DhtDataRequestData {
     #[serde(rename = "_id")]
-    pub msg_id: String,
+    pub request_id: String,
 
     #[serde(rename = "dnaAddress")]
     pub dna_address: Address,
 
+    #[serde(rename = "address")]
+    pub data_address: Address,
+}
+
+/// Data Request from some other agent
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson)]
+pub struct FetchDhtData {
     #[serde(rename = "fromAgentId")]
-    pub from_agent_id: String,
+    pub requester_agent_id: String,
 
-    pub address: String,
+    // DhtDataRequestData
+    #[serde(rename = "_id")]
+    pub request_id: String,
+    #[serde(rename = "dnaAddress")]
+    pub dna_address: Address,
+    #[serde(rename = "address")]
+    pub data_address: Address,
 }
 
+/// Generic DHT data message
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson, Default)]
 pub struct DhtData {
+    #[serde(rename = "dnaAddress")]
+    pub dna_address: Address,
+
+    #[serde(rename = "agentId")]
+    pub provider_agent_id: String,
+
+    #[serde(rename = "address")]
+    pub data_address: Address,
+
+    #[serde(rename = "content")]
+    pub data_content: serde_json::Value,
+}
+
+/// DHT data response from a request
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson, Default)]
+pub struct HandleDhtResultData {
     #[serde(rename = "_id")]
-    pub msg_id: String,
+    pub request_id: String,
+    #[serde(rename = "requesterAgentId")]
+    pub requester_agent_id: String,
 
+    // DhtData
     #[serde(rename = "dnaAddress")]
     pub dna_address: Address,
-
     #[serde(rename = "agentId")]
-    pub agent_id: String,
+    pub provider_agent_id: String,
+    #[serde(rename = "address")]
+    pub data_address: Address,
+    #[serde(rename = "content")]
+    pub data_content: serde_json::Value,
+}
 
-    pub address: String,
-    pub content: serde_json::Value,
+//--------------------------------------------------------------------------------------------------
+// DHT metadata
+//--------------------------------------------------------------------------------------------------
+
+/// Data Request from own p2p-module
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson)]
+pub struct DhtMetaRequestData {
+    pub attribute: String,
+
+    // DhtDataRequestData
+    #[serde(rename = "_id")]
+    pub request_id: String,
+    #[serde(rename = "dnaAddress")]
+    pub dna_address: Address,
```

Comment:
I feel like we might want to add some consistency to the ordering of properties on these structs. Since we almost always have a `request_id` and a `dna_address`, if they were always first, it would be easier to gloss over them when scanning through this file.

(if only we could extend a base struct so we didn't have to duplicate the code)

Reply:
Yeah, I tried to order fields from the bottom, but from the top seems more natural.
We could extend the struct by composition, but that would make things harder on the javascript side, no?

Reply:
just pushed a re-ordering and a renaming of the structs and some fields.

Reply:
I'm good with this. I kinda like the composition idea... since I like to think of this rust protocol as the source-of-truth protocol spec, I don't mind requiring the javascript to conform to whatever's easiest to maintain here.

This PR's rather large already, though, if we want to undertake that change, we may want to do it as a follow-on.

Reply:
Ok, next time there is a big protocol update I'll change to compositions.